packages/@react-aria/slider/src/useSlider.ts

```diff
@@ -115,9 +115,11 @@ export function useSlider(
       }
       let value = state.getPercentValue(percent);
 
+      // Figure out if the click is past all the thumbs to move last thumb when thumbs are stacked.
+      const isPastLastThumb = state.values.every(v => v - value < 0);
       // Only compute the diff for thumbs that are editable, as only they can be dragged
       const minDiff = Math.min(...state.values.map((v, index) => state.isThumbEditable(index) ? Math.abs(v - value) : Number.POSITIVE_INFINITY));
-      const index = state.values.findIndex(v => Math.abs(v - value) === minDiff);
+      const index = isPastLastThumb ? findLastIndex(state.values, v => Math.abs(v - value) === minDiff) : state.values.findIndex(v => Math.abs(v - value) === minDiff);
```

Comment:
I don't think this works for three thumbs where the first two overlap, and the third is further up
I think you might want to replace the Math.min...map... with a reduce or two
What we want is
Closest of the handles to the left, closest of the handles to the right
out of those, whichever side has the closest, if left side scan for highest index of that distance, if right side scan for lowest index of that distance
if both sides have the same distance, then choose left because there is no clear choice

Reply:
I confirmed that this doesn't work for three thumbs and the first two overlapping.